cli/download_source.go

```diff
@@ -0,0 +1,259 @@
+package cli
+
+import (
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/gruntwork-io/terragrunt/config"
+	"os"
+	"github.com/gruntwork-io/terragrunt/errors"
+	"path/filepath"
+	"github.com/hashicorp/go-getter"
+	urlhelper "github.com/hashicorp/go-getter/helper/url"
+	"io/ioutil"
+	"net/url"
+	"fmt"
+)
+
+type TerraformSource struct {
+	// A canonical version of RawSource, in URL format
+	CanonicalSourceURL *url.URL
+
+	// The folder where we should download the source to
+	DownloadDir         string
+
+	// The path to a file in DownloadDir that stores the version number of the code
+	VersionFile         string
+}
```

Comment:
Would be helpful to add a comment on what this struct as a whole represents. It took me a minute to figure out this represents a set of "source code" for Terraform.